fvm/evm/emulator/state/base.go

```diff
@@ -0,0 +1,472 @@
+package state
+
+import (
+	"fmt"
+	"math/big"
+
+	gethCommon "github.com/ethereum/go-ethereum/common"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/onflow/atree"
+
+	"github.com/onflow/flow-go/fvm/evm/types"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+const (
+	// the path where we store the collection ID for accounts
+	AccountsStorageIDKey = "AccountsStorageIDKey"
+	// the path where we store the collection ID for codes
+	CodesStorageIDKey = "CodesStorageIDKey"
+)
+
+// BaseView implements a types.BaseView
+// it acts as the base layer of state queries for the stateDB
+// it stores accounts, codes and storage slots.
+//
+// under the hood it uses a set of collections,
+// one for account's meta data, one for codes
+// and one for each of account storage space.
+type BaseView struct {
```

Comment:
The way BaseView is implemented is not concurrency-safe which I assume is intended usage, if I'm correct maybe add a comment, but if you think of some concurrent usage then we should at least guard the caches

Reply:
I think even if we want to have guards we should do it on the highest level (StateDB)